test/NuGet.Core.FuncTests/Dotnet.Integration.Test/PackCommandTests.cs

```diff
@@ -3966,5 +3991,94 @@ public void PackCommand_WithGeneratePackageOnBuildSet_CanPublish()
+                // Arrange
+                // Set up solution, project, program.cs and package
+                var solution = new SimpleTestSolutionContext(pathContext.SolutionRoot);
+
+                var net461 = NuGetFramework.Parse("net461");
+
+                // Set up project.csproj
+                var projectA = SimpleTestProjectContext.CreateLegacyPackageReference("nonSDKprojectA",
+                                                                                      pathContext.SolutionRoot,
+                                                                                      net461);
+                projectA.Properties.Add("PackageOutputPath", ".\\");
+                projectA.Properties.Add("Authors", "someone");
+                projectA.Save();
+
+                var projectPath = projectA.ProjectPath;
+                var projectFolder = projectPath.Substring(0, projectPath.Length - projectPath.Split('\\').Last().Length - 1);
+
+                using (var stream = new FileStream(projectPath, FileMode.Open, FileAccess.ReadWrite))
+                {
+                    var xml = XDocument.Load(stream);
+
+                    var properties = new Dictionary<string, string>();
+                    properties["Version"] = "4.0.0-aaa";
+                    properties["IncludeAssets"] = "runtime; build; native; contentfiles; analyzers; buildtransitive";
+                    properties["PrivateAssets"] = "all";
+                    ProjectFileUtils.AddItem(xml,
+                                            "PackageReference",
+                                            "NuGet.Build.Tasks.Pack.Sdk2x",
+                                            string.Empty,
+                                            properties,
+                                            new Dictionary<string, string>());
+
+                    ProjectFileUtils.WriteXmlToFile(xml, stream);
+                }
+
+                // Set up program.cs
+                var programContents = @"
+using System;
+
+namespace nonSDKprojectA
+{
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            Console.WriteLine(""Hello World!"");
+        }
     }
+}";
+                var programPath = Path.Combine(projectFolder, "Program.cs");
+                File.WriteAllText(programPath, programContents);
+
+                // Copy the NuGet.Build.Tasks.Pack.Sdk2x package to nupkgSourcePath
+                var nupkgsDirectory = DotnetCliUtil.GetNupkgDirectoryInRepo();
+                var nupkgSourcePath = Directory.GetFiles(nupkgsDirectory, "NuGet.Build.Tasks.Pack.Sdk2x.*").First();
+                var nupkgFileName = nupkgSourcePath.Split("\\").Last();
+                var nupkgDestPath = pathContext.PackageSource + "\\" + nupkgFileName;
+                File.Copy(nupkgSourcePath, nupkgDestPath);
+
+          
+                var msBuildDirectory = "C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\Enterprise\\MSBuild\\Current\\Bin";
```

Comment:
I don't know if this is a good assumption to make. 

This test could start failing 1.5 years down the line when VS 2021 (presumably goes out). 

It might be easier to add an end-to-end test if possible.

Reply:
Since NuGet.Build.Tasks.Pack.Sdk2x is no longer needed, this test will be removed as well. Tracking issue:https://github.com/NuGet/Home/issues/8403